.github/workflows/ci.yml

```diff
@@ -0,0 +1,182 @@
+
+    - name: Install Dependencies
+      if: contains(matrix.os, 'ubuntu') && steps.cache-itk.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        sudo apt update && sudo apt install -y cmake g++ libgl1-mesa-dev libxt-dev libxrender-dev libxext-dev libglu1-mesa-dev make
+
+    - name: Clone ITK
+      if: steps.cache-itk.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        echo "Cloning ITK..."
+        mkdir itk
+        cd itk
+        git init
+        git remote add origin https://github.com/InsightSoftwareConsortium/ITK.git
+        git fetch --depth 1 origin $ITK_HASH
+        git checkout FETCH_HEAD
+
+    - name: Configure ITK
+      if: steps.cache-itk.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        cmake -S itk -B itk-build -DCMAKE_INSTALL_PREFIX=itk-install -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DBUILD_SHARED_LIBS=ON -DBUILD_TESTING=OFF
+
+    - name: Build ITK
+      if: steps.cache-itk.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        cmake --build itk-build --parallel 4 --config ${{ matrix.build_type }}
+
+    - name: Install ITK
+      if: steps.cache-itk.outputs.cache-hit != 'true'
+      shell: bash
+      run: |
+        cmake --install itk-build
+
+  build:
+    ########
+    # BUILD AND TEST IGSIO
+    needs: [update_vtk, update_itk]
+    runs-on: ${{ matrix.os }}
+    strategy:
+      fail-fast: false
+      matrix:
+        os: [ubuntu-latest, windows-latest, macos-latest]
+        build_type: [Release]
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Install Dependencies
+      if: contains(matrix.os, 'ubuntu')
+      shell: bash
+      run: |
+        echo "Installing system dependencies..."
+        sudo apt update && sudo apt install -y cmake g++ libgl1-mesa-dev libxt-dev libxrender-dev libxext-dev libglu1-mesa-dev make
+
+    - name: Restore VTK Cache
+      id: cache-vtk
+      uses: actions/cache/restore@v4
+      with:
+        path: vtk-install
+        key: vtk-${{ runner.os }}-${{ matrix.build_type }}-${{ env.VTK_HASH }}
+
+    - name: Restore ITK Cache
+      id: cache-itk
+      uses: actions/cache/restore@v4
+      with:
+        path: itk-install
+        key: itk-${{ runner.os }}-${{ matrix.build_type }}-${{ env.ITK_HASH }}
+
+    - name: Configure CMake
+      working-directory: ${{github.workspace}}
+      run: |
+        cmake -S ${{github.workspace}} -B ${{github.workspace}}/build -DCMAKE_BUILD_TYPE=${{ matrix.build_type }} -DVTK_DIR='${{github.workspace}}/vtk-install/lib/cmake/vtk' -DITK_DIR='${{github.workspace}}/itk-install/lib/cmake/ITK-5.4' -DIGSIO_SUPERBUILD=ON ${{ env.BUILD_OPTIONS }} -DBUILDNAME="GH Action ${{ matrix.os }} ${{ github.event_name }}"
+
+    - name: Experimental Build
+      if: ${{ github.event_name != 'schedule' }}
+      working-directory: ${{github.workspace}}/build
+      run: |
+        ctest -C ${{ matrix.build_type }} -D Experimental -DCTEST_SITE="GH Action ${{ github.event_name}}"
```

Comment:
Are the build errors not provided in the console output of the job? I was looking at https://github.com/Sunderlandkyl/IGSIO/actions/runs/14040237905/job/39308340892, where I couldn't find specifics of the build error until I matched the run using the site tag (fv-az1050-876) to then find the console output at https://open.cdash.org/viewBuildError.php?buildid=10287094.

Reply:
The verbose output log is fine with me though a link to the CDash results with the specific run filtered out would be nice for the visual that the cdash provides. It was a little difficult to find which of all the runs on the cdash for today corresponded to a specific run. I ended up using the site tag to try and pair between GitHub and CDash. Maybe even instead of "latest pull request" for the build name on CDash it provided the PR number would make it easier to link between the 2.